lawnchair/src/app/lawnchair/ui/preferences/GeneralPreferences.kt

```diff
@@ -116,19 +117,31 @@ fun GeneralPreferences() {
                 destination = subRoute(name = GeneralRoutes.ICON_PACK),
                 subtitle = iconStyleSubtitle,
             )
-            NavigationActionPreference(
-                label = stringResource(id = R.string.icon_shape_label),
-                destination = subRoute(name = GeneralRoutes.ICON_SHAPE),
-                subtitle = iconShapeSubtitle,
-                endWidget = {
-                    IconShapePreview(iconShape = iconShapeAdapter.state.value)
+            ExpandAndShrink(visible = themedIconsEnabled) {
+                SwitchPreference(
+                    adapter = prefs.transparentIconBackground.getAdapter(),
+                    label = stringResource(id = R.string.transparent_background_icons),
+                    description = stringResource(id = R.string.transparent_background_icons_description),
+                )
+            }
+            ExpandAndShrink(visible = !transparentIconBackground.state.value || !themedIconsEnabled) {
+                DividerColumn {
+                    NavigationActionPreference(
+                        label = stringResource(id = R.string.icon_shape_label),
+                        destination = subRoute(name = GeneralRoutes.ICON_SHAPE),
+                        subtitle = iconShapeSubtitle,
+                        endWidget = {
+                            IconShapePreview(iconShape = iconShapeAdapter.state.value)
+                        }
+                    )
+                    SwitchPreference(
+                        adapter = wrapAdaptiveIcons,
+                        label = stringResource(id = R.string.auto_adaptive_icons_label),
+                        description = stringResource(id = R.string.auto_adaptive_icons_description),
+                    )
```

Comment:
I'd recommend *not* hiding these options because not all icons are themed, so removing these options doesn't really make sense. Also some parts rely on the icons shape (like folders, the launch animations when using QS, etc).

If you insist on hiding them, you forgot to nest the slider for Background Lightness

Reply:
Agree , removed hiding this options